### ISSUE TITLE
Add a public constructor to UserAccessToken?

### ISSUE THREAD

**Issue by satu** [robotty/twitch-irc-rs#101]
New to Rust so forgive me if I'm holding it wrong.

I'm implementing a twitch bot for fun and I implemented a basic flow using `RefreshingLoginCredentials`. When obtaining a token for the first time from Twitch API, I have to find a way to pass it to my `CustomTokenStorage`. I couldn't find a way to construct a `UserAccessToken` from within the code:

```rust
#[derive(Debug, Serialize, Deserialize)]
pub struct MyUserAccessToken {
    access_token: String,
    refresh_token: String,
    created_at: DateTime<Utc>,
    expires_at: Option<DateTime<Utc>>,
}


#[tokio::main]
pub async fn main() {
// [...]
    let mut storage = CustomTokenStorage {
        token_checkpoint_file: config.twitch.token_filepath,
    };
    if !args.first_token_file.is_empty() {
        let first_token = fs::read_to_string(args.first_token_file).unwrap();
        let first_token: FirstToken = serde_json::from_str(&first_token).unwrap();
        let created_at = Utc::now();
        let expires_at = created_at + Duration::seconds(first_token.expires_in);
        let user_access_token = MyUserAccessToken {
            access_token: first_token.access_token,
            refresh_token: first_token.refresh_token,
            created_at: created_at,
            expires_at: Some(expires_at),
        };
        let serialized = serde_json::to_string(&user_access_token).unwrap();
        let user_access_token: UserAccessToken = serde_json::from_str(&serialized).unwrap();
        storage.update_token(&user_access_token).await.unwrap();
    }
// [...]
}
```

https://github.com/stuck-overflow/ferris-bot/blob/main/src/main.rs#L239

It would be much simpler if I could construct the `UserAccessToken` with a public constructor. Any reason why it is private?

**Comment by RAnders00** [comment]
That‘s a big oversight on my part. I‘ll make a fix for this tomorrow and will release a bugfix version. Thanks for reporting this!

**Comment by RAnders00** [comment]
Has been released as v2.2.0!
